src/tests/test_api.py

```diff
@@ -157,23 +188,136 @@ def test_addressbook(self):
             []
         )
 
+    def test_subscriptions(self):
+        """Testing the API commands related to subscriptions"""
+        for s in json.loads(self.api.listSubscriptions())['subscriptions']:
+            # special address, added when sqlThread starts
+            if s['address'] == 'BM-GtovgYdgs7qXPkoYaRgrLFuFKz1SFpsw':
+                self.assertEqual(
+                    base64.decodestring(s['label']),
+                    'Bitmessage new releases/announcements')
+                self.assertTrue(s['enabled'])
+                break
+        else:
+            self.fail(
+                'Could not find Bitmessage new releases/announcements'
+                ' in subscriptions')
+        self.assertEqual(
+            self.api.deleteSubscription('BM-GtovgYdgs7qXPkoYaRgrLFuFKz1SFpsw'),
+            'Deleted subscription if it existed.')
+        self.assertEqual(
+            json.loads(self.api.listSubscriptions())['subscriptions'], [])
+
+    def test_send(self):
+        """Test message sending"""
+        # self.api.createDeterministicAddresses(self._seed, 1, 4)
+        addr = self._add_random_address('random_2')
+        msg = base64.encodestring('test message')
+        msg_subject = base64.encodestring('test_subject')
+        ackdata = self.api.sendMessage(
+            'BM-2cWzSnwjJ7yRP3nLEWUV5LisTZyREWSzUK', addr, msg_subject, msg)
```

Comment:
You can use `BM-2cVWtdUzPwF7UNGDrZftWuHWiJ6xxBpiSP`. There is an autoresponder running there, it will send all messages back.

Reply:
I don't think current code is ready for such complex test. To achieve it we should wait for full sync (not sure how to determine it), send a message and wait up to 2 min for reply. And of course also turn off this: https://github.com/Bitmessage/PyBitmessage/blob/v0.6/src/bitmessagemain.py#L231